app/models/facility_group.rb

```diff
@@ -69,4 +80,25 @@ def cohort_analytics(period:, prev_periods:)
   def set_diabetes_management(value)
     facilities.update(enable_diabetes_management: value).map(&:valid?).all?
   end
+
+  def create_region
+    Region.create!(
+      name: name,
+      source: self,
+      reparent_to: state_region,
+      region_type: Region.region_types[:district]
+    )
+  end
+
+  def update_region
+    region.reparent_to = state_region
+    region.name = name
+    region.save!
```

Comment:
Can you check to see if any of these attributes changed to decide if we need to do the update at all here?

Reply:
I think we can avoid that check, just for the sake of reducing complexity. There's already quite a few conditionals to keep in mind. wdyt? https://github.com/simpledotorg/simple-server/pull/1486#discussion_r517017412